prometheus-client/src/Prometheus/Registry.hs

```diff
@@ -19,30 +20,28 @@ import qualified Control.Concurrent.STM as STM
 -- >>> :module +Prometheus
 -- >>> unregisterAll
 
-data RegisteredMetric = forall s. MkRegisteredMetric (Metric s)
-
-type Registry = [RegisteredMetric]
+type Registry = [IO [SampleGroup]]
 
 {-# NOINLINE globalRegistry #-}
 globalRegistry :: STM.TVar Registry
 globalRegistry = unsafePerformIO $ STM.newTVarIO []
 
--- | Registers a metric with the global metric registry.
-register :: Metric s -> IO (Metric s)
-register metric = do
-    let addToRegistry = (MkRegisteredMetric metric :)
-    STM.atomically $ STM.modifyTVar' globalRegistry addToRegistry
+register :: Metric s -> IO s
```

Comment:
This is still public. Could you please document it?

Reply:
Ah, the removal was not intentional, I'll revert that.